targeted/cve-2021-22214.js

```diff
@@ -0,0 +1,150 @@
+/**
+ * Contributed by Astra Security (https://www.getastra.com/)
+ * @author Prince Mendiratta <prince.mendiratta@getastra.com>
+ */
+
+var pluginid = 100024;
+
+// Print Statements using script name
+function logger() {
+    print("[" + this["zap.script.name"] + "] " + arguments[0]);
+}
+
+/**
+ * Unauthenticated Gitlab SSRF - CI Lint API - CVE-2021-22214
+ * 
+ * A function which will be invoked against a specific "targeted" message.
+ * 
+ * @param msg - the HTTP message being acted upon. This is an HttpMessage object.
+ */
+function invokeWith(msg) {
+
+    // Set alert variables
+    var url = msg.getRequestHeader().getURI().toString();
+    var alertName = "Unauthorised SSRF on Gitlab"
+    var alertDesc = "[CVE-2021-22214]\nServer-side request forgery (SSRF) vulnerabilities let an attacker send crafted requests from " +
+        "the back-end server of a vulnerable web application.\n" +
+        "A server-side request forgery vulnerability in GitLab CE/EE affecting all versions starting from 10.5 was possible to exploit for an " +
+        "unauthenticated attacker even on a GitLab instance where registration is limited.\n" +
+        "By exploiting a Server Side Request Forgery vulnerability, attackers may be able to scan the local or external networks to which the " +
+        "vulnerable server is connected to."
+    var alertSol = "Upgrade to the latest version of Gitlab."
+    var alertReference = "https://nvd.nist.gov/vuln/detail/CVE-2021-22214"
+    var cweId = 918; // Server-Side Request Forgery (SSRF)
+    var wascId = 15; // Application Misconfiguration
+
+    // To check if script is running
+    logger("Testing Script against URL - " + url);
+
+    // Initiate the exploit
+    var exploitResponse = sendReq(msg);
+    // The HTTP Response Status Code
+    var status = exploitResponse.getResponseHeader().getStatusCode();
+    // The Content-Type Header
+    // It would ALWAYS be in JSON for the Gitlab Exploit
+    var ctype = exploitResponse.getResponseHeader().getHeader("Content-Type");
+    // The Response Body
+    var rebody = exploitResponse.getResponseBody().toString();
+    
+    // Checks to make sure that the response is by a Gitlab Instance
+    // on the server
+    // A unique term in the demo YAML file which will be present in
+    // the response body when the lint check succeeds
+    if (status === 200 && ctype === "application/json" && rebody.contains("CI_PIPELINE_SOURCE")) {
+        // risk: 0: info, 1: low, 2: medium, 3: high
+        // confidence: 0: falsePositive, 1: low, 2: medium, 3: high, 4: confirmed
```

Comment:
It would be better to keep the variables, given the distance to the actual usage.

Reply:
Have added the comments next to function call.